core/src/main/kotlin/org/jetbrains/kotlinx/dataframe/api/frameCol.kt

```diff
@@ -0,0 +1,653 @@
+
+    /**
+     * @include [FrameColKPropertyDocs] {@arg [CommonFrameColDocs.ReceiverArg] "pathTo"["myColumnGroup"].}
+     */
+    public fun <C> ColumnPath.frameCol(property: KProperty<List<C>>): ColumnAccessor<DataFrame<C>> =
+        asColumnGroup().ensureIsColGroup().frameColumn(property).ensureIsFrameColumn()
+
+    // endregion
+
+    // region index
+
+    /**
+     * @include [CommonFrameColDocs]
+     * {@arg [CommonFrameColDocs.Arg] 0}
+     * {@arg [CommonFrameColDocs.ExampleArg] {@include [CommonFrameColDocs.DoubleExample]}}
+     * @param [index\] The index of the value column.
+     * @throws [IndexOutOfBoundsException\] if the index is out of bounds.
+     */
+    private interface FrameColIndexDocs
+
+    /**
+     * @include [FrameColIndexDocs] {@arg [CommonFrameColDocs.ReceiverArg] `[colsOf][ColumnsSelectionDsl.colsOf]`<`[Int][Int]`>().}
+     * @include [CommonFrameColDocs.FrameColumnTypeParam]
+     * {@arg [CommonFrameColDocs.ExampleArg] {@include [CommonFrameColDocs.SingleExample]}}
+     */
+    @Suppress("INAPPLICABLE_JVM_NAME")
+    @JvmName("ColumnSetDataFrameFrameColIndex")
+    public fun <C> ColumnSet<DataFrame<C>>.frameCol(index: Int): SingleColumn<DataFrame<C>> =
+        getAt(index).ensureIsFrameColumn()
+
+    /**
+     * @include [FrameColIndexDocs] {@arg [CommonFrameColDocs.ReceiverArg] `[colsOf][ColumnsSelectionDsl.colsOf]`<`[Int][Int]`>().}
+     * @include [CommonFrameColDocs.FrameColumnTypeParam]
+     * {@arg [CommonFrameColDocs.ExampleArg] {@include [CommonFrameColDocs.SingleExample]}}
+     */
+    public fun ColumnSet<*>.frameCol(index: Int): SingleColumn<DataFrame<*>> =
+        getAt(index).cast<DataFrame<*>>().ensureIsFrameColumn()
+
+    /**
+     * @include [FrameColIndexDocs] {@arg [CommonFrameColDocs.ReceiverArg]}
+     */
+    @Suppress("INAPPLICABLE_JVM_NAME")
+    @JvmName("frameColUnTyped")
+    public fun ColumnsSelectionDsl<*>.frameCol(index: Int): SingleColumn<DataFrame<*>> =
+        frameCol<Any?>(index)
+
+    /**
+     * @include [FrameColIndexDocs] {@arg [CommonFrameColDocs.ReceiverArg]}
+     * @include [CommonFrameColDocs.FrameColumnTypeParam]
+     */
+    public fun <C> ColumnsSelectionDsl<*>.frameCol(index: Int): SingleColumn<DataFrame<C>> =
+        asSingleColumn().frameCol<C>(index)
+
+    /**
+     * @include [FrameColIndexDocs] {@arg [CommonFrameColDocs.ReceiverArg] myColumnGroup.}
+     */
+    @Suppress("INAPPLICABLE_JVM_NAME")
+    @JvmName("frameColUnTyped")
+    public fun SingleColumn<DataRow<*>>.frameCol(index: Int): SingleColumn<DataFrame<*>> =
+        frameCol<Any?>(index)
+
+    /**
+     * @include [FrameColIndexDocs] {@arg [CommonFrameColDocs.ReceiverArg] myColumnGroup.}
+     * @include [CommonFrameColDocs.FrameColumnTypeParam]
+     */
+    public fun <C> SingleColumn<DataRow<*>>.frameCol(index: Int): SingleColumn<DataFrame<C>> =
+        ensureIsColGroup()
+            .allColumnsInternal()
+            .getAt(index)
+            .cast<DataFrame<C>>()
```

Comment:
I started to think, could this cast not be applied? throw an exception?

Reply:
The cast can never fail since it casts `SingleColumn<*>` to `SingleColumn<DataFrame<C>>`. This type is erased at runtime.

The call below "`.ensureIsFrameColumn()`" performs the check to make sure it is actually a frame column at runtime. It also provides a helpful exception to the user if it isn't a frame column.

In theory, these two functions could be merged if that makes it clearer